dask_cuda/utils.py

```diff
@@ -574,16 +586,60 @@ def parse_device_memory_limit(device_memory_limit, device_index=0, alignment_siz
     def _align(size, alignment_size):
         return size // alignment_size * alignment_size
 
-    if device_memory_limit in {0, "0", None, "auto"}:
+    def parse_fractional(v):
+        """Parse fractional value.
+
+        Ensures ``int(1)`` and ``str("1")`` are not treated as fractionals, but
+        ``float(1)`` is.
+
+        Fractionals must be represented as a ``float`` within the range
+        ``0.0 < v <= 1.0``.
+
+        Parameters
+        ----------
+        v: int, float or str
+            The value to check if fractional.
+
+        Returns
+        -------
+        """
+        # Check if `x` matches exactly `int(1)` or `str("1")`, and is not a `float(1)`
+        is_one = lambda x: not isinstance(x, float) and (x == 1 or x == "1")
+
+        if not is_one(v):
+            with suppress(ValueError, TypeError):
+                v = float(v)
+                if isinstance(v, float) and 0.0 < v <= 1.0:
```

Comment:
Can `isinstance(v, float)` ever be false here?

Reply:
Hmm, I think not, I'm uncertain why I had that added in the first place years ago, probably just a leftover from a previous change. Removed that in https://github.com/rapidsai/dask-cuda/pull/1506/commits/569dc3bba1dbf2689da9cdb837e647f57e50fd62, let's see if that doesn't cause any errors.